src/tribol/mesh/MfemData.hpp

```diff
@@ -0,0 +1,1390 @@
+   */
+  void UpdateRedecomp(redecomp::RedecompMesh& redecomp);
+
+private:
+  /**
+   * @brief Create a finite element space on the (refined) redecomp mesh
+   * 
+   * @param redecomp RCB redecomposed (refined) submesh
+   * @param submesh_fes Original finite element space on the (refined) submesh
+   * @return std::unique_ptr<mfem::FiniteElementSpace> 
+   */
+  static std::unique_ptr<mfem::FiniteElementSpace> CreateRedecompFESpace(
+    redecomp::RedecompMesh& redecomp,
+    mfem::ParFiniteElementSpace& submesh_fes
+  );
+
+  /**
+   * @brief Finite element space on the (coarse) submesh 
+   */
+  mfem::ParFiniteElementSpace& submesh_fes_;
+
+  /**
+   * @brief Finite element space on the (refined) redecomp mesh
+   */
+  mutable std::unique_ptr<mfem::FiniteElementSpace> redecomp_fes_;
+
+  /**
+   * @brief Transfer object between low-order and higher-order grid functions
+   */
+  SubmeshLORTransfer* lor_xfer_;
+
+  /**
+   * @brief Transfer object between (refined) submesh and (refined) redecomp mesh
+   */
+  const redecomp::RedecompTransfer redecomp_xfer_;
+};
+
+/**
+ * @brief Facilitates transferring variables from the original mesh to redecomp
+ * levels
+ *
+ * This class simplifies transferring field variables from/to a grid function on
+ * an mfem::ParMesh to/from a grid function on a redecomp::RedecompMesh.  If
+ * transferring to a low-order grid function on a refined mesh is also required,
+ * this class will perform that transfer as well.
+ */
+class ParentRedecompTransfer
+{
+public:
+  /**
+   * @brief Construct a new ParentRedecompTransfer object
+   *
+   * @param parent_fes Finite element space on the original (coarse) mesh
+   * @param submesh_gridfn Grid function on the (coarse) submesh used to
+   * temporarily store variables being transferred
+   * @param lor_xfer (Optional) Low-order grid function, refined mesh transfer
+   * object
+   * @param redecomp RCB redecomposed (refined) submesh
+   */
+  ParentRedecompTransfer(
+    const mfem::ParFiniteElementSpace& parent_fes,
+    mfem::ParGridFunction& submesh_gridfn,
+    SubmeshLORTransfer* lor_xfer,
+    redecomp::RedecompMesh& redecomp
+  );
+
+  /**
+   * @brief Transfer grid function on original mesh to grid function on redecomp
+   * mesh
+   *
+   * @param src Grid function on (coarse) original mesh
+   * @return mfem::GridFunction on (refined) redecomp mesh
+   */
+  mfem::GridFunction ParentToRedecomp(const mfem::ParGridFunction& src) const;
+  
+  /**
+   * @brief Transfer grid function on redecomp mesh to grid function on original mesh
+   * 
+   * @param src Grid function on (refined) redecomp mesh
+   * @return mfem::ParGridFunction on (coarse) original mesh
+   */
+  mfem::ParGridFunction RedecompToParent(const mfem::GridFunction& src) const;
+
+  /**
+   * @brief Set a new (refined) redecomp mesh associated with the existing (refined) submesh
+   * 
+   * @param redecomp RCB redecomposed (refined) submesh
```

Comment:
Use of the word submesh is confusing here, because I am expecting to see a mfem::ParSubMesh object as input and I don't. So this input argument is a redecomp mesh generated from a RCB redecomposied submesh (and only refined with LOR if the parent was HO)???

Reply:
Method has been removed.